apacheconfig/wloader.py

```diff
@@ -0,0 +1,423 @@
+                   str([self._type] +
+                       [_restore_original(word) for word in self._raw])))
+
+
+class BlockNode(Node):
+    """Creates object containing data for a block.
+
+    Manages any preceding whitespace before the opening block tag, and
+    contains a :class:`apacheconfig.ContentsNode` object representing the
+    block contents.
+    """
+
+    def __init__(self, raw):
+        self._whitespace = ""
+        self._type = raw[0]
+        start = 1
+        if isinstance(raw[start], str) and raw[start].isspace():
+            self._whitespace = raw[start]
+            start += 1
+        self._full_tag = ItemNode(('statement',) + raw[start])
+        self._close_tag = raw[-1]
+        self._contents = None
+        if len(raw[start+1]) > 0:
+            self._contents = ContentsNode(raw[start + 1])
+
+    @staticmethod
+    def parse(raw_str, options={}, parser=None):
+        """Factory for :class:`apacheconfig.BlockNode` by parsing data from a
+        config string.
+
+        Args:
+            options (dict): Additional options to pass to the created parser.
+                Ignored if another ``parser`` is supplied.
+            parser (:class:`apacheconfig.ApacheConfigParser`): optional, to
+                re-use an existing parser. If ``None``, creates a new one.
+        Returns:
+            :class:`apacheconfig.BlockNode` containing data parsed from
+            ``raw_str``.
+        """
+        if not parser:
+            parser = _create_apache_parser(options, start='startitem')
+        return BlockNode(parser.parse(raw_str))
+
+    @property
+    def tag(self):
+        """Returns tag name for this block as a string.
+
+        For instance, ``<block details>\\n</block>`` has tag ``"block"``.
+        """
+        return self._full_tag.name
+
+    @property
+    def arguments(self):
+        """Returns arguments for this block as a literal string.
+
+        For instance, ``<block lots of details>\\n</block>`` has arguments
+         ``"lots of details"``. Can be overwritten.
+        """
+        return self._full_tag.value
+
+    @arguments.setter
+    def arguments(self, arguments):
+        """Sets or overwrites arguments for this block."""
+        self._full_tag.value = arguments
+
+    @property
+    def contents(self):
+        """Returns :class:`apacheconfig.ContentsNode` contained within this
+        block.
+        """
+        return self._contents
+
+    @property
+    def ast_node_type(self):
+        """See base class. Can only return ``"block"`` for
+        :class:`apacheconfig.BlockNode`.
+        """
+        return self._type
+
+    @property
+    def whitespace(self):
+        """Returns preceding whitespace for this node."""
+        return self._whitespace
+
+    @whitespace.setter
+    def whitespace(self):
+        """See base class. Sets preceding whitespace."""
+        return self._whitespace
+
+    def dump(self):
+        """See base class."""
```

Comment:
Would `Node.dump.__doc__` work here?

Reply:
To do `dump.__doc__ = Node.dump.__doc__` ? It also seems that since [Sphinx 1.7](https://github.com/sphinx-doc/sphinx/issues/3140) docstrings are inherited!

Reply:
I meant:

```
    def dump(self):
        __doc__ = Node.dump.__doc__
```

But that's obviously a hack, if Sphinx can respect the inheritance - would it work to just omit the docstring entirely in the overriding `dump()`?